engine/src/main/java/org/hibernate/validator/internal/util/TypeVariableBindings.java

```diff
@@ -69,7 +69,7 @@ else if ( genericSuperType instanceof ParameterizedType ) {
 
 				boolean typeParameterFoundInSubType = false;
 				for ( Entry<TypeVariable<?>, TypeVariable<?>> subTypeParameter : bindings.entrySet() ) {
-					if ( typeArgument.equals( subTypeParameter.getValue() ) ) {
+					if ( typeArgument == subTypeParameter.getValue() ) {
```

Comment:
Just to confirm: are we totally sure we get the same instance when asking for the same argument multiple times? Admittedly, it'd feel safer to stick to equals.